src/OmenAgentResultMapping.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/access/extensions/AccessControlDefaultAdminRules.sol";
+
+/// This contract allows for mapping market addresses to IPFS hashes containing agent results.
+contract OmenAgentResultMapping is AccessControlDefaultAdminRules {
+    bytes32 public constant AGENT_ROLE = keccak256("AGENT_ROLE");
+
+    event AgentResultAdded(address indexed marketAddress, bytes32 ipfsHash, address indexed changer);
+
+    // Mapping of a market address to an array of IPFS hashes (bytes32).
+    mapping(address => bytes32[]) private marketToIPFSHashes;
```

Comment:
With this, we would have a list of trades from Omen and a list of IPFS hashes from this contract, but no way to link them together. 

So, I still think the mapping should be like

```
mapping(<market id> => mapping(<bet id> => ipfs))
```

or the structure the mapping holds could be something like

```
mapping(<market id> => {bet_id: sometype, ipfs: bytes})
```

Reply:
What about having the txHash as a property on IPFS?

Reply:
As you said before, it's not guaranteed to have anything in the IPFS file

Reply:
Having txHash in the PRediction struct solves this thread. Closing this.